index.php

```diff
@@ -1194,6 +1301,32 @@ function startUpdate() {
 			performStep(1, performStepCallbacks[1]);
 		}
 
+		if(document.getElementById('startUpdateButton')) {
+			document.getElementById('startUpdateButton').onclick = function (e) {
+				e.preventDefault();
+				startUpdate();
+			};
+		}
+	</script>
+<?php else: ?>
+	<script>
+		function login() {
+			var xhr = new XMLHttpRequest();
+			xhr.open('GET', window.location.href, true);
+			xhr.setRequestHeader('X-Updater-Auth', document.getElementById('updater-secret-input').value);
+			xhr.onreadystatechange = function () {
+				if (xhr.readyState === 4) {
+					document.getElementsByTagName('html')[0].innerHTML = xhr.responseText;
+					eval(document.getElementsByTagName('script')[0].innerHTML);
```

Comment:
That's kinda a little bit of blackmagic, so what we do here is the following:

1. Load the page via XHR since we need a custom header
2. Insert it via innerHTML
3. Since innerHTML ignores the script tags it gets eval'd manually as well 🙈